planetiler-core/src/main/java/com/onthegomap/planetiler/stream/WriteableJsonStreamArchive.java

```diff
@@ -0,0 +1,231 @@
+ * {@link WriteableJsonStreamArchive.Entry} are separated by newline (by default).
+ */
+public final class WriteableJsonStreamArchive extends WritableStreamArchive {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(WriteableJsonStreamArchive.class);
+
+  /**
+   * exposing meta data (non-tile data) might be useful for most use cases but complicates parsing for simple use cases
+   * => allow to output tiles, only
+   */
+  private static final String OPTION_WRITE_TILES_ONLY = "tiles_only";
+
+  private static final String OPTION_ROOT_VALUE_SEPARATOR = "root_value_separator";
+
+  static final JsonMapper jsonMapper = JsonMapper.builder()
+    .serializationInclusion(Include.NON_ABSENT)
+    .addModule(new Jdk8Module())
+    .addMixIn(TileArchiveMetadata.class, TileArchiveMetadataMixin.class)
+    .addMixIn(Envelope.class, EnvelopeMixin.class)
+    .addMixIn(CoordinateXY.class, CoordinateXYMixin.class)
+    .build();
+
+  private final boolean writeTilesOnly;
+  private final String rootValueSeparator;
+
+  private WriteableJsonStreamArchive(Path p, StreamArchiveConfig config) {
+    super(p, config);
+    this.writeTilesOnly = config.moreOptions().getBoolean(OPTION_WRITE_TILES_ONLY, "write tiles, only", false);
+    this.rootValueSeparator = StreamArchiveUtils.getEscpacedString(config.moreOptions(), TileArchiveConfig.Format.JSON,
+      OPTION_ROOT_VALUE_SEPARATOR, "root value separator", "'\\n'", List.of("\n", " "));
+  }
+
+  public static WriteableJsonStreamArchive newWriteToFile(Path path, StreamArchiveConfig config) {
+    return new WriteableJsonStreamArchive(path, config);
+  }
+
+  @Override
+  protected TileWriter newTileWriter(OutputStream outputStream) {
+    return new JsonTileWriter(outputStream, rootValueSeparator);
+  }
+
+  @Override
+  public void initialize(TileArchiveMetadata metadata) {
+    if (writeTilesOnly) {
+      return;
+    }
+    writeEntryFlush(new InitializationEntry(metadata));
+  }
+
+  @Override
+  public void finish(TileArchiveMetadata metadata) {
+    if (writeTilesOnly) {
+      return;
+    }
+    writeEntryFlush(new FinishEntry(metadata));
+  }
+
+  private void writeEntryFlush(Entry entry) {
+    try (var out = new OutputStreamWriter(getPrimaryOutputStream(), StandardCharsets.UTF_8.newEncoder())) {
+      jsonMapper
+        .writerFor(Entry.class)
+        .withoutFeatures(JsonGenerator.Feature.AUTO_CLOSE_TARGET)
+        .writeValue(out, entry);
+      out.write(rootValueSeparator);
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  private static class JsonTileWriter implements TileWriter {
+
+    private final Writer writer;
+    private final SequenceWriter jsonWriter;
+    private final String rootValueSeparator;
+
+    JsonTileWriter(OutputStream outputStream, String rootValueSeparator) {
+      this.writer = new BufferedWriter(new OutputStreamWriter(outputStream, StandardCharsets.UTF_8.newEncoder()));
```

Comment:
I'm wondering would it be more efficient here to just make a `BufferedOutputStream(outputStrem)` and have jackson `writeValues` go to that? It seems like the curent one does an extra round trip from bytes to utf8 string and back.

Reply:
yes I agree, this makes total sense